HeterogeneousCore/CUDAServices/plugins/CUDAMonitoringService.cc

```diff
@@ -13,6 +13,7 @@
 #include "FWCore/ServiceRegistry/interface/ServiceMaker.h"
 #include "HeterogeneousCore/CUDAServices/interface/CUDAService.h"
 #include "HeterogeneousCore/CUDAUtilities/interface/cudaCheck.h"
+#include "HeterogeneousCore/CUDAUtilities/src/getCachingDeviceAllocator.h"
```

Comment:
does it mean that we should split
```
HeterogeneousCore/CUDAUtilities/src/getCachingDeviceAllocator.h
``` 
into
```
HeterogeneousCore/CUDAUtilities/interface/getCachingDeviceAllocator.h
HeterogeneousCore/CUDAUtilities/src/getCachingDeviceAllocator.cc
```
?

Reply:
This is certainly a sign of deficiencies in the current organization of the caching allocator (stemming from the allocators not being owned by `CUDAService`). I still want to keep the caching allocators hidden. `CUDAService` already includes from the `src`, so I don't think this `#include` would make things worse. On the other hand I could make the status query as part of the public interface, in which case this include-from-src would not be needed. What do you think?

Reply:
I think we should take the occasion and make the caching allocator configurable again ?

Reply:
I moved the device allocator status query to be public. I also added the amount of requested memory (which I think you asked in one email) along the actually allocated memory (and yes, the old settings look horrible).

I'd leave making the allocators configurable to other PR because it would be rather orthogonal to the changes proposed here. And I'd rather do it within https://github.com/cms-patatrack/cmssw/pull/487 (of course if #487 gets much opposition I'm open to change my mind).

Reply:
> I'd leave making the allocators configurable to other PR because it would be rather orthogonal to the changes proposed here.

Sure.